envisage/plugins/ipython_kernel/tests/test_internal_ipkernel.py

```diff
@@ -15,59 +19,106 @@
 
 
 if ipykernel_available:
-    from ipykernel.iostream import IOPubThread
-    from ipykernel.kernelapp import IPKernelApp
+    import ipykernel.iostream
+    import ipykernel.ipkernel
+    import ipykernel.kernelapp
+    import ipykernel.zmqshell
+    import IPython.utils.io
+    import zmq
 
     from envisage.plugins.ipython_kernel.internal_ipkernel import (
         InternalIPKernel)
 
 
+# Machinery for catching logged messages in tests (both for testing
+# the existence and content of those messages, and for hiding the
+# logger output during the test run).
+
+class ListHandler(logging.Handler):
```

Comment:
This feels generally useful, and may belong in `apptools.logging`.

Reply:
Opened enthought/apptools#96